lib/output.js

```diff
@@ -1474,10 +1474,20 @@ function OutputStream(options) {
             output.space();
         }
         if (self.extends) {
+            var parens = self.extends instanceof AST_Binary ||
+                self.extends instanceof AST_Conditional;
             output.print("extends");
-            output.space();
+            if (parens) {
+                output.print("(");
+            } else {
+                output.space();
+            }
             self.extends.print(output);
-            output.space();
+            if (parens) {
+                output.print(")");
+            } else {
+                output.space();
+            }
```

Comment:
I wonder if this could be placed under [`PARENS(AST_Binary)`](https://github.com/mishoo/UglifyJS2/blob/2433bb4e52a5a9be06ed9ac344cc91fafac62ab1/lib/output.js#L651) and a new section of `PARENS(AST_Conditional)`?